backend/corpora/common/utils/secret_config.py

```diff
@@ -47,13 +48,24 @@ def config(self):
 
     def __getattr__(self, name):
         if self.config_is_loaded():
-            return self.value_from_config(name) or self.value_from_env(name) or self.raise_error(name)
+            return (
+                self.value_from_config(name)
+                or self.value_from_env(name)
+                or self.value_from_defaults(name)
```

Comment:
I feel like this could have some unintended consequences if we don't set the vars correctly for a deployment. Like setting a dummy auth and allowing unrestricted access. Are these concerns valid or do we have additional fail-safes or documentation to prevent this?

Reply:
This just adds an additional call to self.value_from_defaults(), which is a no-op for most subclasses of SecretConfig, but CorporaAuthConfig has support to [create oauth default URL's](https://github.com/chanzuckerberg/corpora-data-portal/blob/c351a695ed17c289a3bc54ef131a3a721a32a8b6/backend/corpora/common/corpora_config.py#L41) that follow the previous formatting *if api_authorize_url and api_token_url aren't present in the secret json document* and I've updated the [backend/config/auth0-secret-template.json](https://github.com/chanzuckerberg/corpora-data-portal/pull/673/files#diff-2ae991b600dc1ef0749ef18449e03bc8f9f6f2edcb525d92daca245427584698R3-R5) docs to explain how the defaults are set.

Reply:
i'm sorry (>_<), yeah, I should have seen that.